### ISSUE TITLE
SC2086 in GitHub Actions script

### ISSUE THREAD

**Issue by jokay** [rhysd/actionlint#42]
I have the following **SC2086** violation:

```sh
2021-08-27 20:45:30 [ERROR]   Found errors in [actionlint] linter!
2021-08-27 20:45:30 [ERROR]   Error code: 1. Command output:
------
.github/workflows/release.yml:20:9: shellcheck reported issue in this script: SC2086:info:1:51: Double quote to prevent globbing and word splitting [shellcheck]
   |
20 |         run: |
   |         ^~~~
.github/workflows/release.yml:20:9: shellcheck reported issue in this script: SC2086:info:2:44: Double quote to prevent globbing and word splitting [shellcheck]
   |
20 |         run: |
   |         ^~~~
------
```

for [this GitHub Actions script](https://github.com/jokay/docker-ccu-historian/blob/main/.github/workflows/release.yml#L20) which sets [environment variables](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable).

```sh
- name: Set variables
  run: |
    echo "CREATED=$(date -u +%Y-%m-%dT%H:%M:%S%z)" >> $GITHUB_ENV
    echo "VERSION=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
```

any advice on get this one fixed would be appreciated.

**Comment by shogo82148** [comment]
SC2086 is a warning code of shellcheck, go to https://github.com/koalaman/shellcheck/wiki/SC2086 for more detail.

Due to implementation limitations, actionlint does not report the correct location for warnings in scripts.
Using shell checks directly is helpful.

```
$ cat foo.sh
#!/bin/bash

echo "CREATED=$(date -u +%Y-%m-%dT%H:%M:%S%z)" >> $GITHUB_ENV
echo "VERSION=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV

$ shellcheck foo.sh

In foo.sh line 3:
echo "CREATED=$(date -u +%Y-%m-%dT%H:%M:%S%z)" >> $GITHUB_ENV
                                                  ^---------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
echo "CREATED=$(date -u +%Y-%m-%dT%H:%M:%S%z)" >> "$GITHUB_ENV"


In foo.sh line 4:
echo "VERSION=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
                                           ^---------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
echo "VERSION=${GITHUB_REF#refs/tags/}" >> "$GITHUB_ENV"

For more information:
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```

**Comment by jokay** [comment]
Thx, will change it accordingly.

**Comment by rhysd** [comment]
Thanks @shogo82148 for the follow up.

Small additional info:

> SC2086:info:1:51:

This means shellcheck pointed line:1, column:51 in the script at `run:`.
